oak_functions/loader/src/lookup.rs

```diff
@@ -38,9 +40,14 @@ impl LookupData {
     ///
     /// The returned instance is empty, and must be populated by calling the [`LookupData::refresh`]
     /// method at least once.
-    pub fn new_empty(lookup_data_url: &str, logger: Logger) -> LookupData {
+    pub fn new_empty(
+        lookup_data_url: &str,
+        use_gcp_metadata_auth: bool,
```

Comment:
I find the bool not very readable when invoking this fn; another option would be to actually pass in a fn to fetch the token here, and we can either pass a reference to `get_access_token`, or to antoher fn `empty_token` instead that always returns None or empty string. Up to you anyways.

Reply:
I agree that the bool is not very clear, but storing async function references was not great either. So I decided to swtich to using an enum, which hopefully makes it more readable.

Reply:
LGTM; anyways the fact that the fn is async is irrelevant in terms of ref I think, it's just a normal fn that returns `Future<Out>`, right? Or does it not work because of lifetimes and traits? (just curious, no need to spend more time)

Reply:
Yes, lifetimes and traits made it more complex. I could make it work with using `BoxFuture`, a closure and pinning, but it made the code look much more complicated.